src/docs/codelabs/extended-flex-widget.md

```diff
@@ -0,0 +1,597 @@
+
+`CrossAxisAlignment.center`
+: Positions children at the middle of the cross axis. (`Row` only)
+
+`CrossAxisAlignment.stretch`
+: Stretches children across the cross axis. 
+(Top-to-bottom for `Row`, left-to-right for `Column`)
+
+`CrossAxisAlignment.baseline`
+: Aligns children by their character baselines. (`Text` class only, and requires that the `textBaseline` property is set to `TextBaseline.alphabetic`. See the [Text class](#text-class) section for an example.)
+
+#### Example: Modify cross axis alignment
+{:.no_toc}
+{{site.alert.secondary}}
+  The following example explicitly sets the `crossAxisAlignment` property to its default value,
+  `CrossAxisAlignment.center`. 
+
+  To demonstrate cross axis alignment, `Row` now contains a `BiggerBlueBox` widget
+  that is taller than the other widgets, and `mainAxisAlignment` is set to
+ `MainAxisAlignment.spaceAround`. 
+
+  **1.** Click the **Run** button.
+
+  **2.** Change `CrossAxisAlignment.center` to `CrossAxisAlignment.start`, and run again. 
+{{site.alert.end}}
+{% comment %}
+  Gist: https://gist.github.com/70a6eb88f13019eec349a57bc4fd5fe0
+{% endcomment %}
+<iframe src="https://dartpad.dev/experimental/embed-new-flutter.html?id=70a6eb88f13019eec349a57bc4fd5fe0&amp;theme=dark&amp;split=60" width="100%" height="400px"></iframe>
+{{site.alert.tip}}
+  Before moving to the next section, change `CrossAxisAlignment.start` to one of the other values. 
+{{site.alert.end}}
+
+## Flexible widget
+
+As you've seen, the `mainAxisAlignment` and `crossAxisAlignment` properties determine 
+how `Row` and `Column` position widgets along both axes. 
+`Row` and `Column` first lay out widgets of a fixed size.
+Fixed size widgets are considered *inflexible* because they can't resize
+themselves after they've been laid out.
+
+The `Flexible` widget wraps a widget, so the widget becomes resizable. 
+When the `Flexible` widget wraps a widget, the widget becomes the `Flexible` widget's child 
+and is considered *flexible*. 
+After inflexible widgets are laid out,
+flexible widgets are resized according to their `flex` and `fit` properties.:
+
+`flex`
+: Compares itself against other `flex` properties before determining 
+what fraction of the total extra space each `Flexible` widget receives.
+
+`fit` 
+: Determines whether a `Flexible` widget fills all of its extra space.
+
+#### Example: Testing flex values
+{:.no_toc}
+{{site.alert.secondary}}
+  In the following example, two `Flexible` widgets wrap two `BlueBox` widgets. 
+  The `Flexible` widgets contain `flex` properties with values set to 1 (the default value).
+
+  When `flex` properties are compared, 
+  the ratio between the values determines 
+  what fraction of the total extra space each `Flexible` widget receives.
+
+  In this example, the sum of the `flex` values (2),
+  determines that each `Flexible` widget increases
+  by half of the free space. 
+  The fixed-width widget remains the same size.
+
+  **1.** Click the **Run** button. 
```

Comment:
It's not great to have a one-item numbered list. And wait — this isn't really a numbered list?! Why not?

Nonetheless, I think this warrants a 2nd item:
Change the width of the UI by dragging the divider between the code and the UI. <describe what to look for>